vendor/deps/cli-ui/lib/cli/ui.rb

```diff
@@ -210,6 +207,29 @@ def self.enable_color=(bool)
 
     self.enable_color = $stdout.tty?
 
+    # Check whether emojis are enabled in Formatter output. This depends on
+    # the OS running the CLI UI, since Windows does not fully support them yet
+    # See https://github.com/microsoft/terminal/issues/190 for more info.
+    #
+    def self.enable_emoji?
+      @enable_emoji
+    end
+
+    # Determine which OS is running [:mac, :linux, :windows]
+    #
+    # ==== Returns
+    # A symbol indicating the current OS
+    def self.os
+      require 'rbconfig'
+
+      host = RbConfig::CONFIG["host"]
+      return :mac if /darwin/.match(host)
```

Comment:
Perhaps this could return an object to encapsulate OS specifics, e.g.

```ruby
class Cli::OS::Mac
  def supports_emoji?
    true
  end
end

class Cli::OS::Windows
  def supports_emoji?
    false
  end
end
```